tasks/eva_2835/gather_per_species_per_assembly_variant_remapping_counts.py

```diff
@@ -0,0 +1,270 @@
+        idx = []
+
+        for key in all_keys[row - 1]:
+            position = all_columns.index(key)
+            idx.append(position)
+
+        # Populating the values for the other stats columns for a particular iteration
+        for col in range(len(all_columns)):
+            if col in idx:
+                key = all_columns[col]
+                position = all_keys[row - 1].index(key)
+                stats = all_values[row - 1][position]
+                worksheet.write(row, col + 2, stats, stats_fmt)
+
+            else:
+                worksheet.write(row, col + 2, 0, stats_fmt)
+
+        # Incrementing the row to write the next record
+        row = row + 1
+
+    # Closing the workbook
+    workbook.close()
+
+
+def gather_counts_per_tax_per_assembly(path, taxid, assembly_accession):
+    """
+    This function is used to store the counts of the remapped variants along wih the reason for failures
+    at different rounds (i.e. for different lengths of the flanking region) for a particular taxonomy and
+    assembly
+
+    Input: The taxonomy id and the assembly accession from generate_output() along with the input files full
+           path
+
+    Output: All the statistics in global lists
+
+    """
+
+    # Setting the filename for the eva counts
+    filename_eva = str(assembly_accession) + "_eva_remapped_counts.yml"
+    filename_eva = os.path.join(path, str(taxid), str(assembly_accession), "eva", filename_eva)
+
+    # Setting the filename for the dbsnp counts
+    filename_dbsnp = str(assembly_accession) + "_dbsnp_remapped_counts.yml"
+    filename_dbsnp = os.path.join(path, str(taxid), str(assembly_accession), "dbsnp", filename_dbsnp)
+
+    # Calling functions for gathering stats per file from eva and dbsnp data
+    keys_values = gather_counts_per_file(filename_eva)
+    keys_values_dbsnp = gather_counts_per_file(filename_dbsnp)
+
+    # Adding the values of dbsnp and eva with the common keys
+    for key in keys_values_dbsnp:
+        if key in keys_values:
+            keys_values_dbsnp[key] = keys_values_dbsnp[key] + keys_values[key]
+
+    # Concatenating the two dictionaries
+    keys_values_dbsnp = {**keys_values, **keys_values_dbsnp}
+
+    # Sorting the dictionary using the keys
+    keys_values_dbsnp = {key: value for key, value in sorted(keys_values_dbsnp.items())}
+
+    # Splitting the dictionary into two lists - one for keys and one for values
+    keys, values = zip(*keys_values_dbsnp.items())
+    keys = list(keys)
+    values = list(values)
+
+    # Adding the keys and values list to the master list for each taxonomy and assembly
+    all_keys.append(keys)
+    all_values.append(values)
+
+    # Adding the taxonomy and assembly accession for a particular file
+    all_tax_id.append(taxid)
+    all_assembly.append(assembly_accession)
+
+    # Updating the final output columns
+    global all_columns
+    all_columns.extend(keys)
+    all_columns = list(dict.fromkeys(all_columns)
+    all_columns.sort()
```

Comment:
All these don't really belong in this function. 
It should be extracted out to the top level function.

Reply:
The code is reorganized and the suggestions are incorporated